### PR TITLE
[IOTDB-1968] Mount the Metric Service as a Built-in IService

### PR DESCRIPTION
**Pull request by stormbroken**
I remove old metric service, and migrate new metric framework into iotdb

### CODE REVIEW

File: metrics/micrometer-metrics/src/main/java/org/apache/iotdb/metrics/micrometer/reporter/MicrometerPrometheusReporter.java [hunk 1]
```diff
@@ -65,11 +65,10 @@ public boolean start() {
             .route(
```
Comment (xinzhongtianxia) [inline metrics/micrometer-metrics/src/main/java/org/apache/iotdb/metrics/micrometer/reporter/MicrometerPrometheusReporter.java:65]: Last month, there is a feature merged into master, it's about providing rest apis for users.
The rest service will also startup a http server, can these two http servers merged into one?

File: server/src/main/java/org/apache/iotdb/db/service/MetricsService.java [hunk 2]
```diff
@@ -18,53 +18,93 @@
  */
 package org.apache.iotdb.db.service;
```
Comment (SteveYurongSu) [inline server/src/main/java/org/apache/iotdb/db/service/MetricsService.java:19]:
```suggestion
package org.apache.iotdb.db.service.metrics;
```

File: server/src/main/java/org/apache/iotdb/db/service/MetricsServiceMBean.java [hunk 3]
```diff
@@ -22,8 +22,6 @@
 
 public interface MetricsServiceMBean {
```
Comment (SteveYurongSu) [inline server/src/main/java/org/apache/iotdb/db/service/MetricsServiceMBean.java:23]: Move it to package org.apache.iotdb.db.service.metrics;

File: docs/UserGuide/System-Tools/Metric-Tool.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    Unless required by applicable law or agreed to in writing,
+    software distributed under the License is distributed on an
+    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+    KIND, either express or implied.  See the License for the
+    specific language governing permissions and limitations
+    under the License.
+
+-->
+
+# Metric Tools
```
Comment (SteveYurongSu) [inline docs/UserGuide/System-Tools/Metric-Tool.md:22]: We should add its reference to site/src/main/.vuepress/config.js to update the website.

**Review by SteveYurongSu — COMMENTED** [review]
It's pretty good overall :D

One more question, do exposed ports in the docker environment need to be modified?

File: metrics/micrometer-metrics/src/main/java/org/apache/iotdb/metrics/micrometer/reporter/MicrometerPrometheusReporter.java (hunk 1 above)
Comment (stormbroken) [reply metrics/micrometer-metrics/src/main/java/org/apache/iotdb/metrics/micrometer/reporter/MicrometerPrometheusReporter.java:65]: ok, it's a good idea. I think we could finish this in another PR.

File: server/src/main/java/org/apache/iotdb/db/service/MetricsService.java (hunk 2 above)
Comment (stormbroken) [reply server/src/main/java/org/apache/iotdb/db/service/MetricsService.java:19]: ok, I will fix this.

File: server/src/main/java/org/apache/iotdb/db/service/MetricsServiceMBean.java (hunk 3 above)
Comment (stormbroken) [reply server/src/main/java/org/apache/iotdb/db/service/MetricsServiceMBean.java:23]: ok, I will fix this.

File: docs/UserGuide/System-Tools/Metric-Tool.md (hunk 4 above)
Comment (stormbroken) [reply docs/UserGuide/System-Tools/Metric-Tool.md:22]: ok, I will fix this

**Comment by stormbroken** [comment]
yes, there is need to expose port for docker, but notice that you need also to use -p to expose port when run container

File: metrics/micrometer-metrics/src/main/java/org/apache/iotdb/metrics/micrometer/reporter/MicrometerPrometheusReporter.java (hunk 1 above)
Comment (stormbroken) [reply metrics/micrometer-metrics/src/main/java/org/apache/iotdb/metrics/micrometer/reporter/MicrometerPrometheusReporter.java:65]: We will do this on this issue: https://issues.apache.org/jira/projects/IOTDB/issues/IOTDB-2127?filter=reportedbyme

File: site/src/main/.vuepress/config.js [hunk 5]
```diff
@@ -738,7 +738,8 @@ var config = {
 							['System-Tools/MLogParser-Tool','MLogParser Tool'],
 							['System-Tools/NodeTool','Node Tool'],
 							['System-Tools/Query-History-Visualization-Tool','Query History Visualization Tool'],
-							['System-Tools/Watermark-Tool','Watermark Tool']
+							['System-Tools/Watermark-Tool','Watermark Tool'],
+							['System-Tools/Metric-Tool','Metric-Tool']
```
Comment (SteveYurongSu) [inline site/src/main/.vuepress/config.js:742]:
```suggestion
							['System-Tools/Metric-Tool','Metric Tool']
```

**Pull request merged**